npm/pkg/controlplane/controllers/v2/namespacecontroller.go

```diff
@@ -359,14 +363,16 @@ func (nsc *NamespaceController) syncUpdateNamespace(newNsObj *corev1.Namespace)
 	addToIPSets, deleteFromIPSets := util.GetIPSetListCompareLabels(curNsObj.LabelsMap, newNsLabel)
 	// Delete the namespace from its label's ipset list.
 	for _, nsLabelVal := range deleteFromIPSets {
-		labelKey := util.GetNSNameWithPrefix(nsLabelVal)
 
 		labelKeySet := &ipsets.IPSetMetadata{Name: nsLabelVal, Type: ipsets.KeyLabelOfNamespace}
-		toBeAdded := []*ipsets.IPSetMetadata{ipsets.NewIPSetMetadata(newNsName, ipsets.Namespace)}
+		if util.IsKeyLabelSetName(nsLabelVal) {
```

Comment:
This function name is misleading.
It seems like `IsKeyValueLabelSetName` is more proper.
If this is right, there are other places to be fixed (e.g., podController)

Reply:
Changed the name to IsKeyValueSetName, this function check if the setname is of type Key or Key:Value